File: snapchat/profile/oc/CodiaDemoOC/src/ProfileImageNetwork.m
//
//  ProfileImageNetwork.m
//
//  Created by codia-figma
//

#import <UIKit/UIKit.h>
#import <Masonry/Masonry.h>
#import "ProfileImageNetwork.h"

@interface ProfileImageNetwork ()

/// background view
@property (nonatomic, strong) UIView *customView1;
/// background view
@property (nonatomic, strong) UIView *customView2;
/// background view
@property (nonatomic, strong) UIView *customView3;
/// background view
@property (nonatomic, strong) UIView *customView4;

@end

@implementation ProfileImageNetwork

- (instancetype)initWithFrame:(CGRect)frame {
    if (self = [super initWithFrame:frame]) {
        [self setupSubviews];
        [self setupConstraints];
        self.backgroundColor = UIColor.clearColor;
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];
}


- (void)setupSubviews {
    [self addSubview:self.customView1];
    [self addSubview:self.customView4];
    [self addSubview:self.customView3];
    [self addSubview:self.customView2];
}

- (void)setupConstraints {
    [self.customView1 mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.mas_left);
        make.width.mas_equalTo(4);
        make.bottom.equalTo(self.mas_bottom).offset(-2);
        make.height.mas_equalTo(16);
    }];

    [self.customView2 mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.mas_left);
        make.width.mas_equalTo(4);
        make.bottom.equalTo(self.customView1.mas_bottom);
        make.height.mas_equalTo(4);
    }];

    [self.customView3 mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.customView1.mas_right).offset(2);
        make.width.mas_equalTo(4);
        make.bottom.equalTo(self.customView1.mas_bottom);
        make.height.mas_equalTo(8);
    }];

    [self.customView4 mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.customView3.mas_right).offset(2);
        make.width.mas_equalTo(4);
        make.bottom.equalTo(self.customView1.mas_bottom);
        make.height.mas_equalTo(12);
    }];
}




- (UIView *)customView1 {
    if (!_customView1) {
        _customView1 = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 4, 16)];
        _customView1.backgroundColor = [UIColor colorWithRed:0.77 green:0.77 blue:0.77 alpha:0.41];
    }
    return _customView1;
}

- (UIView *)customView2 {
    if (!_customView2) {
        _customView2 = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 4, 4)];
        _customView2.backgroundColor = [UIColor colorWithRed:1.00 green:1.00 blue:1.00 alpha:1.00];
    }
    return _customView2;
}

- (UIView *)customView3 {
    if (!_customView3) {
        _customView3 = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 4, 8)];
        _customView3.backgroundColor = [UIColor colorWithRed:1.00 green:1.00 blue:1.00 alpha:1.00];
    }
    return _customView3;
}

- (UIView *)customView4 {
    if (!_customView4) {
        _customView4 = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 4, 12)];
        _customView4.backgroundColor = [UIColor colorWithRed:1.00 green:1.00 blue:1.00 alpha:1.00];
    }
    return _customView4;
}

@end

File: snapchat/profile/swift/CodiaDemoSwift/src/ProfileImageNetwork.swift
//
//  ProfileImageNetwork.swift
//
//  Created by codia-figma
//

import UIKit
import SnapKit


class ProfileImageNetwork: UIView {

    /// background view
    lazy var customView1: UIView = {
        let view = UIView(frame: CGRect(x: 0, y: 0, width: 4, height: 16))
        view.backgroundColor = UIColor(red: 0.77, green: 0.77, blue: 0.77, alpha: 0.41)
        return view
    }()

    /// background view
    lazy var customView2: UIView = {
        let view = UIView(frame: CGRect(x: 0, y: 0, width: 4, height: 4))
        view.backgroundColor = UIColor(red: 1.00, green: 1.00, blue: 1.00, alpha: 1.00)
        return view
    }()

    /// background view
    lazy var customView3: UIView = {
        let view = UIView(frame: CGRect(x: 0, y: 0, width: 4, height: 8))
        view.backgroundColor = UIColor(red: 1.00, green: 1.00, blue: 1.00, alpha: 1.00)
        return view
    }()

    /// background view
    lazy var customView4: UIView = {
        let view = UIView(frame: CGRect(x: 0, y: 0, width: 4, height: 12))
        view.backgroundColor = UIColor(red: 1.00, green: 1.00, blue: 1.00, alpha: 1.00)
        return view
    }()

    override init(frame: CGRect) {
        super.init(frame: frame)
        setupSubviews()
        setupConstraints()
        self.backgroundColor = UIColor.clear
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    func layoutSubview() {
      super.layoutSubviews()
    }

    func setupSubviews() {
        self.addSubview(self.customView1)
        self.addSubview(self.customView4)
        self.addSubview(self.customView3)
        self.addSubview(self.customView2)
    }

    func setupConstraints() {
        customView1.snp.makeConstraints { make in
            make.left.equalTo(self.snp.left)
            make.width.equalTo(4)
            make.bottom.equalTo(self.snp.bottom).offset(-2)
            make.height.equalTo(16)
        }

        customView2.snp.makeConstraints { make in
            make.left.equalTo(self.snp.left)
            make.width.equalTo(4)
            make.bottom.equalTo(customView1.snp.bottom)
            make.height.equalTo(4)
        }

        customView3.snp.makeConstraints { make in
            make.left.equalTo(customView1.snp.right).offset(2)
            make.width.equalTo(4)
            make.bottom.equalTo(customView1.snp.bottom)
            make.height.equalTo(8)
        }

        customView4.snp.makeConstraints { make in
            make.left.equalTo(customView3.snp.right).offset(2)
            make.width.equalTo(4)
            make.bottom.equalTo(customView1.snp.bottom)
            make.height.equalTo(12)
        }
    }
}